workflow/rules/frag_to_tagAlign.smk

```diff
@@ -18,18 +18,11 @@ rule frag_to_tagAlign:
 	shell:
 		"""
 		# Make, sort and compress tagAlign file from fragment file
-		gunzip -c {input.frag_file} | \
-		awk 'BEGIN{{OFS="\\t"}} {{
-			mid = int(($2 + $3) / 2); # Calculate middle point
-			# Repeat the output tags according to the 5th column of the fragment file
-			for(i = 1; i <= $5; i++) {{
-				print $1, $2, mid, "N", 1000, "+"; # positive strand region
-				print $1, mid + 1, $3, "N", 1000, "-"; # negative strand region
-			}}
-		}}' | \
-		bedtools sort -faidx {params.chrom_sizes} -i stdin | \
-		bgzip > {output.tagAlign_sort_file}
-		
+		LC_ALL=C zcat {input.frag_file}  | sed '/^#/d' | \
+		awk -v OFS='\t' '{{mid=int(($2+$3)/2); print $1,$2,mid,"N",1000,"+"; print $1,mid+1,$3,"N",1000,"-"}}' | \
+		sort -k 1,1V -k 2,2n -k3,3n --parallel 5 | \
```

Comment:
We should add a `resources` section to this rule so that we can request many cpus when submitting this as a job to slurm